lib/spack/llnl/util/filesystem.py

```diff
@@ -1097,16 +1097,26 @@ def visit_directory_tree(root, visitor, rel_path='', depth=0):
     for f in dir_entries:
         if sys.version_info >= (3, 5, 0):
             rel_child = os.path.join(rel_path, f.name)
-            islink, isdir = f.is_symlink(), f.is_dir()
+            islink = f.is_symlink()
+            # Windows issue with reading symlinked archive
+            # results in [WinError 1921]
```

Comment:
(question) Does `isdir` always fail on symlinks? Or when you say archive you mean for a symlink to a file that is for example a `.zip`?

Reply:
`isdir` will fail on symlinks to a file that is a `.zip` or something similar.

Reply:
Isn't that a Python bug?

Reply:
On further investigation, this may be due to how broken or self referencing symlinks are handled on Windows, rather than a bug.

Reply:
We spoke about updating the comment here to provide more context on when things fail on Windows